tests/suites/test_suite_nist_kw.function

```diff
@@ -229,75 +232,125 @@ void nist_kw_ciphertext_lengths(int in_len, int out_len, int mode, int res)
 exit:
     mbedtls_free(ciphertext);
     mbedtls_free(plaintext);
-    mbedtls_nist_kw_free(&ctx);
+    psa_destroy_key(key_id);
+    PSA_DONE();
 }
 /* END_CASE */
 
 /* BEGIN_CASE */
-void mbedtls_nist_kw_wrap(int cipher_id, int mode, data_t *key, data_t *msg,
+void mbedtls_nist_kw_wrap(int mode, const data_t *key, data_t *msg,
                           data_t *expected_result)
 {
-    unsigned char result[528];
-    mbedtls_nist_kw_context ctx;
-    size_t result_len, i, padlen;
+    mbedtls_svc_key_id_t key_id = MBEDTLS_SVC_KEY_ID_INIT;
+    size_t result_length, padlen;
 
-    mbedtls_nist_kw_init(&ctx);
+    padlen = (msg->len % 8 != 0) ? 8 - (msg->len % 8) : 0;
+    int result_size = msg->len + 8 + padlen;
+    unsigned char *result;
+    result = mbedtls_calloc(1, result_size);
```

Comment:
```suggestion
    unsigned char *result = NULL;
    TEST_CALLOC(result, result_size);
```
Allocating memory can fail! Also `TEST_CALLOC` is consistent about a length of 0 (always `NULL`, whereas the C standard leaves it implementation-defined).